src/util/diproche/diprocheInterface.ts

```diff
@@ -0,0 +1,87 @@
+import IssueCode from "../../checking/issueCodes";
+import { getAllIssues } from "../../vocabularyChecker/detectWrongSyntax";
+import modes from "./predicateList.json";
+/**
+ * Since the user input most likely lacks the correct predicate,
+ * it needs to be added. For easier maintainability in case
+ * of an expanding predicate list, the corresponding predicate is
+ * added with respect to the current @param mode. This can, for
+ * example correspond to the concept of "Spielwiese". The user should
+ * be able to just stay inside a mode and not have to add this mode
+ * himself. Hence, the mode needs to be read out from the browser.
+ */
+export function addPredicate(userInput: string, mode: string) {
+	for (const predicate of modes.predicates) {
+		if (predicate.valueOf() === mode.valueOf()) {
+			return replaceModeWithPredicate(mode).concat("(").concat(userInput).concat(").");
+		}
+	}
+}
+
+export function replaceModeWithPredicate(mode: string) {
+	if (mode === "propositionalLogic") {
+		return "diproche";
+	}
+	if (mode === "firstOrderPredicateLogic") {
+		return "diproche_fo";
+	}
+	if (mode === "test") {
+		return "teste";
+	}
+	return "InvalidPredicate";
+}
```

Comment:
If your `mode` were a union type (or an enum, which provides union-functionality), you could use exhaustive checking, see https://www.typescriptlang.org/docs/handbook/advanced-types.html Exhaustiveness checking.

Reply:
Do you mean that I could define it as string | undefined?

Reply:
Actually, `"propositionalLogic" | "firstOrderPredicateLogic" | "test"` is a type, too (one that can only take the values `"propositionalLogic"`,`"firstOrderPredicateLogic"` or `"test"`). You can give it a shorter name, too:
```typescript
type Mode = "propositionalLogic" | "firstOrderPredicateLogic" | "test";

export function replaceModeWithPredicate(mode: Mode): string {
```

If you do so, you do not have to return `InvalidPredicate` at the end, since the compiler can determine this line is never reached. It is good practice to throw a `UnexpectedError` there though, so we know about totally unexpected inputs as soon as possible:

```Typescript
export class UnexpectedError: Error {
  constructor(unexpected: never) {
    super(`Unexpected value '${unexpected}'!`);
  }
}
```

`never` is the Typescript type that a variable gets when previous checks ruled out all possible values. This usually only occurs when someone tampered with the typesystem (by using raw Javascript, casting or `any`).

Reply:
Understood. But which superconstructor should get called? I thought, superconstructors need to have some superclass as reference.

Reply:
I just pushed the code as far as I understood you. One test now doesn't work anymore (which is understandable) and I get an error because of super (which I don't know how to solve currently).

Reply:
The tip by @TimothyGillespie did it ;)